common/validators/image_validator.py

```diff
@@ -0,0 +1,98 @@
+from collections.abc import Mapping
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext_lazy as _
+from django.utils.deconstruct import deconstructible
+from PIL import Image
+
+
+@deconstructible
+class ImageValidator:
+    default_messages = {
+        "dimensions": _(
+            "Image dimensions should not exceed %(max_width)s x %(max_height)s pixels."
+        ),
+        "size": _(
+            "File size exceeds the allowed limit of %(max_size)s KiB."
+        ),
+        "invalid_image": _(
+            "The uploaded file is not a valid image."
+        ),
+        "invalid_extension": _(
+            "Invalid file extension. Only PNG, JPG, and JPEG are allowed."
+        ),
+    }
+
+    def __init__(self, max_size=None, max_width=None, max_height=None, messages=None):
+        """
+        Initialize the image validator with optional size and dimension limits.
+
+        :param max_size: Maximum file size in bytes
+        :param max_width: Maximum width in pixels
+        :param max_height: Maximum height in pixels
+        :param messages: Custom error messages
+        """
+        self.max_size = max_size
+        self.max_width = max_width
+        self.max_height = max_height
+        if messages is not None and isinstance(messages, Mapping):
+            self.messages = {**self.default_messages, **messages}
+        else:
+            self.messages = self.default_messages
+
+    def __call__(self, value):
+        # Validate file extension
+        valid_extensions = ('png', 'jpg', 'jpeg')
+        if not value.name.lower().endswith(valid_extensions):
+            raise ValidationError(
+                self.messages['invalid_extension'],
+                code='invalid_extension'
+            )
```

Comment:
Instead of checking the file extension from the name, consider validation the image using the file's content type, which can be more secure. For instance, you can use the `mimetype`s module or check `image.format` in the `PIL.Image` object

Reply:
Done